internal/updater/updater.go

```diff
@@ -173,9 +173,24 @@ func (u *UpdateInstaller) prepareInstall(installTargetPath string, args []string
 	return installerPath, args, nil
 }
 
-func (u *UpdateInstaller) InstallBlocking(installTargetPath string, args ...string) error {
+func (u *UpdateInstaller) InstallBlocking(installTargetPath string, args ...string) (rerr error) {
```

Comment:
`InstallBlocking` only addresses the `state update` flow IIRC. For eg. `install.sh` it uses `InstallWithProgress`, so either we need the same changes there or we need to handle these at a higher level.

Reply:
@Naatan I have been operating under that assumption that we're lacking in update events for install, download, etc., **not** bare install events. The document link in the ticket links to the "Updates" section.

Reply:
@mitchell-as that was the original reason for enriching a lof of these events, yeah. But that doesn't change the fact that the events in question are for the installer. We need to be able to rely on these events behaving as we expect them to. Having them only fire on certain use-cases would violate that expectation.